File: Periodo4/Desenvolvimento de Aplicativos Moveis/imcApp/src/model/Pessoa.js
class Pessoa {
    constructor(nome,peso, altura){
        this.nome = nome;
        this.peso = peso;
        this.altura = altura;

    }
    imc(){
        return this.peso/(this.altura*this.altura)
    }

    faixa(){
//         18.50 – 24.99: Peso Normal.
// 25.00 – 29.99: Pré-Obesidade.
// 30.00 – 34.99: Obesidade Grau I.
// 35.00 – 39.99: Obesidade Grau II.
// ≥40.00: Obesidade Grau III.
        let imc = this.imc();
        if(imc < 18.5){
            return "magro";
        }else if(imc <=29.99){
            return "Normal"
        }else if(imc<=34.99){
            return "Obesidade Grau 1"
        }else if(imc<=39.99){
            return "Obesidade Grau 2"
        }else{
            return "Obesidade Grau 3"
        }
    }
}

export default Pessoa;

File: Periodo4/Testes e Manutencao de Software/projetoAula/src/main/java/com/example/demo/model/Pessoa.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Pessoa {

    private String nome;
    private double peso,altura;

    public double imc() throws Exception {

        if (peso <= 0){
            throw new Exception("peso não pode ser menor que zero");
        }
        if (altura <= 0){
            throw new Exception("altura não pode ser menor que zero");
        }

        return peso/(altura*altura);
    }

//    18.50 – 24.99: Peso Normal.
//            25.00 – 29.99: Pré-Obesidade.
//30.00 – 34.99: Obesidade Grau I.
//35.00 – 39.99: Obesidade Grau II.
//≥40.00: Obesidade Grau III.
    public String faixa() throws Exception {
        double fx = imc();
        if (fx <= 18.5){
            return "magro";
        }else if (fx <= 24.99){
            return "Normal";
        }else if (fx <= 29.99){
            return "Pré-Obesidade";
        }else if (fx <= 34.99){
            return "Obesidade Grau I";
        }else if (fx <= 39.99){
            return "Obesidade Grau II";
        }else{
            return "Obesidade Grau III";
        }

    }

}
